wgpu-hal/src/gles/device.rs

```diff
@@ -603,8 +603,8 @@ impl crate::Device for super::Device {
 
         unsafe { gl.bind_buffer(target, None) };
 
-        if !is_coherent && desc.usage.contains(crate::BufferUses::MAP_WRITE) {
-            map_flags |= glow::MAP_FLUSH_EXPLICIT_BIT;
+        if !is_coherent {
+            map_flags |= glow::MAP_FLUSH_EXPLICIT_BIT | glow::MAP_WRITE_BIT;
```

Comment:
issue: CI is rejecting this change, though I don't understand why.

Reply:
It's not great but that's the current requirement of the buffer initialization code.

Maybe we should only do it if `MAPPABLE_PRIMARY_BUFFERS` is enabled. If it's not, I think we can avoid flushing altogether as the buffer in question can only have `MAP_READ` & `COPY_DST` usages.

Thoughts? @Wumpf @cwfitzgerald

Reply:
Actually we need to do this since the GL backend doesn't support `MAPPABLE_PRIMARY_BUFFERS` and now the WebAssembly CI job is failing.

Reply:
Okay. Well, I'm inclined to trust the collectively invoked judgment of other contributors in this matter; once CI passes, I'm happy. 🤷🏻‍♂️